From: Mitchell Augustin <mitchell.augustin@canonical.com>
Date: Mon, 11 Aug 2025 14:21:40 -0500
Subject: status/network: Fix quicksettings enterprise connections

In d6779f34dac6c80e178a308314fe59884a65f383, the switch
to shell_app_activate_action() was made without
also wrapping the action parameter in an av variant.
This patch adjusts network.js to use the correct av
variant type when attempting to connect to an
enterprise AP from the quicksettings panel.

Fixes d6779f34dac6c80e178a308314fe59884a65f383
Closes https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8590

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3834>
---
 js/ui/status/network.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/status/network.js b/js/ui/status/network.js
index 466ec61..e8a384e 100644
--- a/js/ui/status/network.js
+++ b/js/ui/status/network.js
@@ -64,8 +64,8 @@ function ssidToLabel(ssid) {
 }
 
 function launchSettingsPanel(panel, ...args) {
-    const param = new GLib.Variant('(sav)',
-        [panel, args.map(s => new GLib.Variant('s', s))]);
+    const param = new GLib.Variant('av',
+        [new GLib.Variant('(sav)', [panel, args.map(s => new GLib.Variant('s', s))])]);
 
     const app = Shell.AppSystem.get_default()
         .lookup_app('org.gnome.Settings.desktop');
